modules/core/02-client/keeper/keeper.go

```diff
@@ -38,11 +38,11 @@ type Keeper struct {
 }
 
 // NewKeeper creates a new NewKeeper instance
-func NewKeeper(cdc codec.BinaryCodec, key storetypes.StoreKey, router *types.Router, legacySubspace types.ParamSubspace, sk types.StakingKeeper, uk types.UpgradeKeeper) Keeper {
+func NewKeeper(cdc codec.BinaryCodec, key storetypes.StoreKey, legacySubspace types.ParamSubspace, sk types.StakingKeeper, uk types.UpgradeKeeper) Keeper {
```

Comment:
niice, I like not having to pass the router in

Reply:
aye!! there's possibly more improvements to be made on structuring and what not but let's go with this for now

Reply:
Could also instantiate the storeProvider here when creating the router.